src/senaite/jsonapi/catalog.py

```diff
@@ -64,12 +65,22 @@ def __call__(self, query):
     def __repr__(self):
         return "<Catalog %s>" % self.get_catalog().getId()
 
-    def get_catalog(self, default="portal_catalog"):
+    def get_catalog(self, default="uid_catalog"):
         name = req.get("catalog")
         if not name:
-            # get the mapped catalog for the given portal_type
-            catalogs = senaiteapi.get_catalogs_for(self.portal_type)
-            name = catalogs[0].getId() if len(catalogs) > 0 else default
+            catalogs = []
+            for portal_type in to_list(self.portal_type):
+                # get the mapped catalog for the given portal_type
+                mapped_catalogs = senaiteapi.get_catalogs_for(portal_type)
```

Comment:
The default `portal_catalog` in `get_catalogs_for` is probably used by purpose, because during the setup/installation of a new site we don't have yet the `uid_catalog` in place. But passing through the `default` is something I'm going to change.